win32/src/gballoc_ll_jemalloc.c

```diff
@@ -214,3 +224,284 @@ void gballoc_ll_print_stats(void)
+                                break;
+                            }
+                        }
+                    }
+                }
+
+                if (i < narenas)
+                {
+                    for (uint32_t j = 0; j < i; j++)
+                    {
+                        snprintf_result = snprintf(command, sizeof(command), "arena.%" PRIu32 ".dirty_decay_ms", j);
+                        if (snprintf_result < 0 || (size_t)snprintf_result >= sizeof(command))
+                        {
+                            LogError("snprintf(command=%p, sizeof(command)=%zu, arena.<j>.dirty_decay_ms, j=%" PRIu32 ") failed %s during cleanup", command, sizeof(command), j, (snprintf_result < 0) ? " with encoding error" : " due to insufficient buffer size");
+                        }
+                        else
+                        {
+                            arena_decay_result = je_mallctl(command, NULL, NULL, &old_decay_milliseconds, sizeof(old_decay_milliseconds));
+                            if (arena_decay_result != 0)
+                            {
+                                if (arena_decay_result == EFAULT)
+                                {
+                                    LogError("je_mallctl(command=%s, NULL, NULL, &old_decay_milliseconds=%p, sizeof(old_decay_milliseconds)=%zu) failed during cleanup as the arenas doesn't exist", command, &old_decay_milliseconds, sizeof(old_decay_milliseconds));
+                                }
+                                else
+                                {
+                                    LogError("je_mallctl(command=%s, NULL, NULL, &old_decay_milliseconds=%p, sizeof(old_decay_milliseconds)=%zu) failed during cleanup", command, &old_decay_milliseconds, sizeof(old_decay_milliseconds));
+                                }
+                            }
+                        }
+                    }
+                    result = MU_FAILURE;
+                }
+                else
+                {
+                    result = 0;
+                    LogInfo("jemalloc dirty_decay_ms set to %" PRId64 ", old value was %" PRId64 "", decay_milliseconds, old_decay_milliseconds);
+                    goto all_ok;
+                }
+            }
+            
+            if (je_mallctl("arenas.dirty_decay_ms", NULL, NULL, &old_decay_milliseconds, sizeof(old_decay_milliseconds)) != 0)
+            {
+                LogError("je_mallctl(arenas.dirty_decay_ms, NULL, NULL, &old_decay_milliseconds=%p, sizeof(old_decay_milliseconds)=%zu) failed during cleanup", &old_decay_milliseconds, sizeof(old_decay_milliseconds));
+            }
+        }
+    }
+
+all_ok:
+    return result;
+}
+
+static int gballoc_ll_set_muzzy_decay(int64_t decay_milliseconds)
```

Comment:
>gballoc_ll_set_muzzy_decay [](http://example.com/codeflow?start=11&length=26)

muzzy and dirty functions are SAME except the spelling of muzzy/decay here and there.

Any chance we collapse the 2 functions in just 1? Halve the chances of a bug with half the code!

Reply:
Merged the two